src/ceph-volume/ceph_volume/api/lvm.py

```diff
@@ -1352,30 +1360,114 @@ def create_lvs(volume_group, parts=None, size=None, name_prefix='ceph-lv'):
 #
 ###########################################################
 
-PV_FIELDS = 'pv_name,pv_tags,pv_uuid,vg_name,lv_uuid'
-VG_FIELDS = 'vg_name,pv_count,lv_count,snap_count,vg_attr,vg_size,vg_free,vg_free_count'
-LV_FIELDS = 'lv_tags,lv_path,lv_name,vg_name,lv_uuid,lv_size'
+def convert_filters_to_str(filters):
+    """
+    Convert filter args from dictionary to following format -
+        filters={filter_name=filter_val,...}
+    """
+    if not filters:
+        return filters
+
+    filter_arg = ''
+    for k, v in filters.items():
+        filter_arg += k + '=' + v + ','
+    # get rid of extra comma at the end
+    filter_arg = filter_arg[:len(filter_arg) - 1]
+
+    return filter_arg
+
+def convert_tags_to_str(tags):
+    """
+    Convert tags from dictionary to following format -
+        tags={tag_name=tag_val,...}
+    """
+    if not tags:
+        return tags
+
+    tag_arg = 'tags={'
+    for k, v in tags.items():
+        tag_arg += k + '=' + v + ','
+    # get rid of extra comma at the end
+    tag_arg = tag_arg[:len(tag_arg) - 1] + '}'
 
-def get_pvs(fields=PV_FIELDS, sep='";"', filters=''):
+    return tag_arg
+
+def make_filters_lvmcmd_ready(filters, tags):
+    """
+    Convert filters (including tags) from dictionary to following format -
+        filter_name=filter_val...,tags={tag_name=tag_val,...}
+
+    The command will look as follows =
+        lvs -S filter_name=filter_val...,tags={tag_name=tag_val,...}
+    """
+    filters = convert_filters_to_str(filters)
+    tags = convert_tags_to_str(tags)
+
+    if filters and tags:
+        return filters + ',' + tags
+    if filters and not tags:
+        return filters
+    if not filters and tags:
+        return tags
+    else:
+        return ''
+
+def get_pvs(fields=PV_FIELDS, sep='";"', filters='', tags=None):
+    filters = make_filters_lvmcmd_ready(filters, tags)
     args = ['pvs', '--no-heading', '--readonly', '--separator=' + sep, '-S',
             filters, '-o', fields]
 
     stdout, stderr, returncode = process.call(args, verbose_on_failure=False)
     pvs_report = _output_parser(stdout, fields)
     return [PVolume(**pv_report) for pv_report in pvs_report]
 
-def get_vgs(fields=VG_FIELDS, sep='";"', filters=''):
+# TODO: rename to get_lv when the current get_lv is deleted
+def _get_pv(fields=PV_FIELDS, sep='";"', filters=None, tags=None):
```

Comment:
We need to find a better name. An underscore in python implies by convention a private method.

`get_first_pv()`?

Reply:
As the comment says, I wanted to name it get_pv, but since that name is occupied I used `_get_pv` temporarily. Anyways, `get_first_pv` sounds better.

Reply:
Using `get_first_pv` now, let me know if you prefer `get_lv`.